### ISSUE TITLE
Huobi: The coin pair does not currently offer subscription services

### ISSUE THREAD

**Issue by Tongoko** [ccxt/ccxt#20131]
### Operating System

Windows

### CCXT Version

4.1.66

### Description

On Huobi, I'm receiving a error for a smaller altcoin that "the coin pair does not currently offer subscription services".
On the previous version 4.0.18, this error didn't show up.

Has anything been changed? Is there a workaround for this?

Thanks!

### Code

```
ccxt.base.errors.BadRequest: {"status":"error","ts":1701004351078,"id":"1","err-code":"bad-request","err-msg":"The coin pair does not currently offer subscription services."}

```

**Comment by sc0Vu** [comment]
Hi @Tongoko 

Can you post verbose log and your code? BTW, the error was from huobi exchange, are you sure it still work now in 4.0.18?

**Comment by Tongoko** [comment]
@sc0Vu 
Yes, I downgraded ccxt to 4.0.18 and it's working again.

How do I do verbose log?

**Comment by carlosmiei** [comment]
Hello @Tongoko, you can do this:
```Python
await exchange.loadMarkets() # load markets first to avoid polluting the log
exchange.verbose = True
# Your call here
```

**Comment by Tongoko** [comment]
@sc0Vu @carlosmiei 
Is this what you are looking for? Verbose log:

2023-11-28T11:50:43.052Z connecting to wss://api.huobi.pro/feed with timeout 10000 ms
2023-11-28T11:50:43.067Z connected
2023-11-28T11:50:43.067Z ping loop
2023-11-28T11:50:43.067Z sending {'sub': 'market.(altcoin)usdt.mbp.20', 'id': '1'}
2023-11-28T11:50:43.067Z pong WSMessage(type=<WSMsgType.PONG: 10>, data=bytearray(b''), extra='')
2023-11-28T11:50:43.083Z message {"status":"error","ts":1701172243055,"id":"1","err-code":"bad-request","err-msg":"The coin pair does not currently offer subscription services."}


As mentioned earlier, after downgrading to CCXT 4.0.18, it's working normally again.


```
await self.exchange.loadMarkets()
self.exchange.verbose = True
order_book = await self.exchange.watchOrderBook(self.symbol)
```

**Comment by carlosmiei** [comment]
@Tongoko yes and besides that, what's `self.symbol` here

**Comment by Tongoko** [comment]
@carlosmiei self.symbol is the altcoin pair here, for example DIO/USDT

**Comment by carlosmiei** [comment]
@Tongoko Thanks we will take a look and let you know.

**Comment by carlosmiei** [comment]
@Tongoko Yeah the issue is related to the default limit, some coins don't accept all the limits. Can you try to provide limit = 150?

**Comment by Tongoko** [comment]
@carlosmiei Great, yes that solved it!

**Comment by Tongoko** [comment]
@carlosmiei Actually, after a couple of runs, the limit=150 solves the issue, but now has a tendency to cause synchronization errors. Is there an other way around this bug?

**Comment by Tongoko** [comment]
Okay I now added 'options': {'watchOrderBook': {'maxRetries': 100}}, which seems to solve the synchronization issue

**Comment by carlosmiei** [comment]
@Tongoko Yeah I don't have a better suggestion at the moment.

**Comment by carlotestor** [comment]
Just to close the loop on this one for anyone finding it later — the error comes from HTMX/Huobi rejecting the *orderbook depth* rather than the pair itself. Low-liquidity altcoins don't offer every `mbp` (market-by-price) subscription level, and the default limit CCXT used changed between 4.0.18 and 4.1.x, which is why the downgrade "fixed" it.

Working configuration:

```python
exchange = ccxt.pro.huobi({
    'options': {
        'watchOrderBook': {'maxRetries': 100},
    },
})
orderbook = await exchange.watch_order_book(symbol, 150)
```

`limit=150` (or 20/400 depending on the pair) picks a level the pair actually supports, and the raised `maxRetries` handles the resync churn you saw afterwards. Wrapping the `watch_order_book` call in a try/except inside your loop is also recommended so a transient resync doesn't kill the task.
